### PR TITLE
dco-foundation pipeline

### PR DESCRIPTION
**Pull request by bburky**
Updated to use OIDC to connect to access AWS: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services

### CODE REVIEW

**Comment by bburky** [comment]
Trust policy for `arn:aws:iam::***:role/NapsDevS3`, currently limited to just this specific repo. The StringLike can be updated to authorize additional repos.

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::***:oidc-provider/token.actions.githubusercontent.com"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringLike": {
                    "token.actions.githubusercontent.com:sub": "repo:naps-dev/dco-foundation:*"
                },
                "ForAllValues:StringEquals": {
                    "token.actions.githubusercontent.com:aud": "sts.amazonaws.com",
                    "token.actions.githubusercontent.com:iss": "https://token.actions.githubusercontent.com"
                }
            }
        }
    ]
}
```

File: zarf.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      - "registry1.dso.mil/ironbank/opensource/jaegertracing/jaeger-operator:1.35.0"
+      - "registry1.dso.mil/ironbank/opensource/jaegertracing/jaeger-es-index-cleaner:1.35.2"
+      - "registry1.dso.mil/ironbank/opensource/jaegertracing/all-in-one:1.35.2"
+      - "registry1.dso.mil/ironbank/opensource/jaegertracing/jaeger-agent:1.35.2"
+      - "registry1.dso.mil/ironbank/opensource/jaegertracing/jaeger-ingester:1.35.2"
+      - "registry1.dso.mil/ironbank/opensource/jaegertracing/jaeger-query:1.35.2"
+      - "registry1.dso.mil/ironbank/opensource/jaegertracing/jaeger-collector:1.35.2"
+      
+
+  - name: big-bang-core-plg-assets
```
Comment (blancharda) [inline zarf.yaml:98]: Is there a motivation for including the PLG assets if we're deploying EFK?

File: kustomizations/bigbang/values.yaml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  enabled: true
+  ingressGateways:
+    public-ingressgateway:
+      type: "LoadBalancer"
+
+  gateways:
+    public:
+      tls:
+        key: |
+          -----BEGIN PRIVATE KEY-----
```
Comment (blancharda) [inline kustomizations/bigbang/values.yaml:31]: Is this the bigbang.dev cert?

File: zarf.yaml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    repos:
+      - https://repo1.dso.mil/platform-one/big-bang/apps/sandbox/loki.git@3.0.5-bb.4
+      - https://repo1.dso.mil/platform-one/big-bang/apps/sandbox/promtail.git@4.2.0-bb.2
+      - https://repo1.dso.mil/platform-one/big-bang/apps/sandbox/tempo.git@0.15.1-bb.7
+    images:
+      - registry1.dso.mil/ironbank/opensource/grafana/loki:2.5.0
+      - registry1.dso.mil/ironbank/opensource/grafana/promtail:v2.5.0
+      - registry1.dso.mil/ironbank/opensource/grafana/tempo-query:1.4.1
+      - registry1.dso.mil/ironbank/opensource/grafana/tempo:1.4.1
+
```
Comment (bburky) [inline zarf.yaml:110]: @mikevanhemert I assume we can drop this if we're not using PLG?

```suggestion
```

File: kustomizations/bigbang/values.yaml (hunk 2 above)
Comment (bburky) [reply kustomizations/bigbang/values.yaml:31]: Yes.... and I think it's expired?
```
Issuer: C=US, O=Let's Encrypt, CN=R3
Validity
    Not Before: Mar 15 08:15:08 2022 GMT
    Not After : Jun 13 08:15:07 2022 GMT
Subject: CN=*.bigbang.dev
```

Not sure how to best handle keys long term, I think anything is fine for now.

Comment (blancharda) [reply kustomizations/bigbang/values.yaml:31]: I'm planning on updating the bigbang version after we merge this... so keeping it in plaintext for now (outdated and all) is probably fine.

We should brainstorm on longterm solutions though for when we have a real cert / other secrets..

**Pull request merged**